### PR TITLE
Implement process type specific env vars

### PR DESCRIPTION
**Pull request by jabrown85**
Include process type specific paths when processing env vars during launch.

Spec: https://github.com/buildpacks/spec/pull/98

Signed-off-by: Jesse Brown <jabrown85@gmail.com>

### CODE REVIEW

File: launch/launcher.go [hunk 1]
```diff
@@ -106,7 +106,18 @@ func (l *Launcher) env() error {
 			if err := l.Env.AddEnvDir(filepath.Join(path, "env")); err != nil {
 				return err
 			}
-			return l.Env.AddEnvDir(filepath.Join(path, "env.launch"))
+			if process.Type != "" {
+				if err := l.Env.AddEnvDir(filepath.Join(path, "env", process.Type)); err != nil {
+					return err
+				}
+			}
```
Comment (ekcasey) [inline launch/launcher.go:113]: Based on my reading of the RFC, I thought we were only accepting `env.launch/<process>/` and not accepting `env/<process>`

Comment (jabrown85) [reply launch/launcher.go:113]: I don't see an issue with removing `env/<process>`. cc @jkutner is that your understanding as well? If so, I'll move things around.

Comment (jabrown85) [reply launch/launcher.go:113]: As far as implementing `profile` scripts, I'll wait for that same question to be answered and follow the same decision. This PR will aim to match that particular spec PR.

Comment (jabrown85) [reply launch/launcher.go]: I actually went ahead and made the change specific to `env.launch/<process>`. I think the RFC is a bit ambiguous but having `env/<process>` would only add confusion without making the feature much easier to use.

**Comment by jabrown85** [comment]
@ekcasey I went ahead and pushed the `profile.d/:process` changes as discussed. Please re-review at your convenience!

**Comment by jabrown85** [comment]
> Looks good! I think we should add a process-specific profile script to the acceptance tests before merging though

I was afraid you'd say that 😅  - I'll see about adding one for the new process specific path and old path. There doesn't seem to be any acceptance tests for `profile.d`.

**Comment by jabrown85** [comment]
@ekcasey process specific profile.d script acceptance test is up.

**Pull request merged**